### PR TITLE
feat: Update kratos chart for v6+

### PR DESCRIPTION
**Pull request by Demonsthere**
## Related issue
- Implements https://github.com/ory/k8s/pull/290

## Proposed changes
- Add courier statefulset
- Adjust values

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

File: .circleci/values/kratos.yaml [hunk 1]
```diff
@@ -53,7 +53,7 @@ kratos:
     hashers:
       argon2:
         parallelism: 1
-        memory: 131072
+        memory: 2048MB
```
Comment (aeneasr) [inline .circleci/values/kratos.yaml:56]:
```suggestion
        memory: 128MB
```

File: helm/charts/kratos/templates/deployment.yaml [hunk 2]
```diff
@@ -70,14 +73,14 @@ spec:
             "--dev",
             {{- end}}
             "--config",
-            "/etc/config/kratos.yaml"
+            "/etc/config/meta/kratos.yaml"
```
Comment (aeneasr) [inline helm/charts/kratos/templates/deployment.yaml:76]:
```suggestion
            "/etc/config/kratos.yaml"
```

File: helm/charts/kratos/templates/deployment.yaml [hunk 3]
```diff
@@ -70,14 +73,14 @@ spec:
...
           ]
           volumeMounts:
             {{- if .Values.deployment.extraVolumeMounts }}
 {{ toYaml .Values.deployment.extraVolumeMounts | indent 12 }}
             {{- end }}
             - name: {{ include "kratos.name" . }}-config-volume
-              mountPath: /etc/config
+              mountPath: /etc/config/meta
```
Comment (aeneasr) [inline helm/charts/kratos/templates/deployment.yaml:83]:
```suggestion
              mountPath: /etc/config
```

File: helm/charts/kratos/templates/statefulset-mail.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        app.kubernetes.io/instance: {{ .Release.Name }}      
+    spec:
+      containers:
+        - name: {{ include "kratos.fullname" . }}-courier
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args: [ "courier", "watch", "--config", "/etc/config/meta/kratos.yaml" ]
+          volumeMounts:
+            - name: {{ include "kratos.fullname" . }}-config-volume
+              mountPath: /etc/config/meta
```
Comment (aeneasr) [inline helm/charts/kratos/templates/statefulset-mail.yaml:38]:
```suggestion
              mountPath: /etc/config
```

File: helm/charts/kratos/templates/statefulset-mail.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "kratos.fullname" . }}-courier
+        app.kubernetes.io/instance: {{ .Release.Name }}      
+    spec:
+      containers:
+        - name: {{ include "kratos.fullname" . }}-courier
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args: [ "courier", "watch", "--config", "/etc/config/meta/kratos.yaml" ]
```
Comment (aeneasr) [inline helm/charts/kratos/templates/statefulset-mail.yaml:35]:
```suggestion
          args: [ "courier", "watch", "--config", "/etc/config/kratos.yaml" ]
```

**Review by aeneasr — CHANGES_REQUESTED** [review]

**Comment by aeneasr** [comment]
If it passes the CI it means it works right? 😅

**Comment by Demonsthere** [comment]
right now the CI makes sure the chart can be installed, and that the applications are running afterwards (containers are not ins a crashloop, but running instead) 
Possible improvements:
- [ ] create some sort of healthcheck for the courier

**Review by aeneasr — APPROVED** [review]

**Pull request merged**